src/modules/app/containers/Switcher.js

```diff
@@ -72,71 +57,48 @@ export class Switcher extends React.Component<PropsType> {
-      case CATEGORIES_ROUTE:
-        return Switcher.renderFailureLoadingComponents([citiesPayload, categoriesPayload]) ||
-          <CategoriesPage categories={categoriesPayload.data} cities={citiesPayload.data} />
-      case EVENTS_ROUTE:
-        return Switcher.renderFailureLoadingComponents([citiesPayload, eventsPayload]) ||
-          <EventsPage cities={citiesPayload.data} events={eventsPayload.data} />
-      case EXTRAS_ROUTE:
-        return Switcher.renderFailureLoadingComponents([citiesPayload, extrasPayload]) ||
-          <ExtrasPage cities={citiesPayload.data} extras={extrasPayload.data} />
-      case SPRUNGBRETT_ROUTE:
-        return Switcher.renderFailureLoadingComponents([citiesPayload, extrasPayload, sprungbrettJobsPayload]) ||
-          <SprungbrettExtraPage cities={citiesPayload.data} extras={extrasPayload.data}
-                                sprungbrettJobs={sprungbrettJobsPayload.data} />
-      case WOHNEN_ROUTE:
-        return Switcher.renderFailureLoadingComponents([citiesPayload, extrasPayload, wohnenPayload]) ||
-          <WohnenExtraPage cities={citiesPayload.data} extras={extrasPayload.data} offers={wohnenPayload.data} />
-      case DISCLAIMER_ROUTE:
-        return Switcher.renderFailureLoadingComponents([citiesPayload, disclaimerPayload]) ||
-          <DisclaimerPage cities={citiesPayload.data} disclaimer={disclaimerPayload.data} />
-      case SEARCH_ROUTE:
-        return Switcher.renderFailureLoadingComponents([citiesPayload, categoriesPayload]) ||
-          <SearchPage cities={citiesPayload.data} categories={categoriesPayload.data} />
-      case POIS_ROUTE:
-        return Switcher.renderFailureLoadingComponents([citiesPayload, poisPayload]) ||
-          <PoiPage cities={citiesPayload.data} pois={poisPayload.data} />
-      case NOT_FOUND:
-        // The only possibility to be in the NOT_FOUND route is if we have "/:param" as path and the param is neither
-        // "disclaimer" nor a city, so we want to show an error that the param is not an available city
-        if (param) {
-          const error = new CityNotFoundError({city: param})
-          return <FailureSwitcher error={error} />
-        }
+    if (currentRoute === NOT_FOUND) {
+      // The only possibility to be in the NOT_FOUND route is if we have "/:param" as path and the param is neither
+      // "disclaimer" nor a city, so we want to show an error that the param is not an available city
+      if (param) {
+        const error = new CityNotFoundError({city: param})
+        return <FailureSwitcher error={error} />
+      }
     }
-    throw new Error(`No content was selected for the route ${currentRoute} to be displayed.`)
+    const RouteContent = getRouteContent(currentRoute)
+    const payloads = getRouteConfig(currentRoute).getRequiredPayloads(allPayloads)
+    return Switcher.renderFailureLoadingComponents(payloads) ||
+      <>
+        <Helmet getPageTitle={getRouteConfig(currentRoute).getPageTitle} />
```

Comment:
(2)
give Helmet a `title` prop instead of a `getPageTitle` and add description and languageChangePaths likewise:
```js
<Helmet title={getRouteConfig(currentRoute).getPageTitle(t, location, payloads)} />
```